pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -880,6 +917,11 @@ func deviceInfoTask(ctxPtr *zedagentContext, triggerDeviceInfo <-chan struct{})
 		case <-stillRunning.C:
 		}
 		agentlog.StillRunning(agentName + "devinfo")
+
+		// mark self as ready to receive publish trigger again
+		if !ctxPtr.devInfoTaskReady {
```

Comment:
The if check may be redundant? Always mark yourself ready ??? You are setting it to true in any case?